src/tools/kdb/colors.c

```diff
@@ -0,0 +1,26 @@
+/**
+ * @file
+ *
+ * @brief Implementation of ANSI escape helper functions in the kdb tool
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <colors.h>
+#include <kdberrors.h>
+#include <stdio.h>
+
+const char * formatString (char ** buffer, int colorMode, const char * escapeCode, const char * text)
+{
+	if (colorMode == CLI_COLOR_NEVER) return text;
+
+	size_t len = elektraStrLen (escapeCode) + elektraStrLen (text) + elektraStrLen (RESET) + 1;
```

Comment:
You don't need the `+ 1` at the end, `elektraStrLen` already includes the zero terminator (i.e. `elektraStrLen (X) == strlen (X) + 1`).

Reply:
done